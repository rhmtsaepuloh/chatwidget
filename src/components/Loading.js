import React from "react";

import "../styles/Loading.scss";

const Loading = ({ display }) => {
  return (
    <div
      id='myModal'
      className='modal'
      style={{ display: display ? "block" : "none", borderRadius: "20px" }}>
      <div className='flex items-center h-full'>
        <div className='bg-white w-32 mx-auto px-4 rounded py-4'>
          <div className='lds-ring mx-auto'>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
          </div>
          <p className='text-center text-sm'>Memuat...</p>
        </div>
      </div>
    </div>
  );
};

export default Loading;
