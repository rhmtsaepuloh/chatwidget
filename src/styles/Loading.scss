/* The Modal (background) */
.modal {
  position: fixed; /* Stay in place */
  z-index: 100; /* Sit on top */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgb(0, 0, 0); /* Fallback color */
  background-color: rgba(0, 0, 0, 0.4); /* Black w/ opacity */
  textarea:focus,
  input:focus {
    outline: none;
  }
  .lds-ring {
    width: 80px;
    height: 80px;
    margin: auto;
  }
  .lds-ring div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 64px;
    height: 64px;
    margin: 8px;
    border: 6px solid #fa565d;
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #fa565d transparent transparent transparent;
  }
  .lds-ring div:nth-child(1) {
    animation-delay: -0.45s;
  }
  .lds-ring div:nth-child(2) {
    animation-delay: -0.3s;
  }
  .lds-ring div:nth-child(3) {
    animation-delay: -0.15s;
  }
  @keyframes lds-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
}
