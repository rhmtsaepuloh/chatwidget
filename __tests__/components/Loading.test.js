/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import Loading from "../../src/components/Loading";

describe("render correctly loading components", () => {
  test("renders display is true", () => {
    const wrapper = shallow(<Loading display={true} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders display is false", () => {
    const wrapper = shallow(<Loading display={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
