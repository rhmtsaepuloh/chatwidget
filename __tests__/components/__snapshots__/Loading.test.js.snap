// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly loading components renders display is false 1`] = `ShallowWrapper {}`;

exports[`render correctly loading components renders display is true 1`] = `ShallowWrapper {}`;
